src/sentry/notifications/activity/base.py

```diff
@@ -15,30 +16,30 @@
 from sentry.utils.linksign import generate_signed_link
 
 
-class ActivityEmail:
-    def __init__(self, activity):
+class ActivityNotification:
+    def __init__(self, activity: Activity) -> None:
         self.activity = activity
         self.project = activity.project
         self.organization = self.project.organization
         self.group = activity.group
 
-    def _get_subject_prefix(self):
+    def _get_subject_prefix(self) -> str:
         prefix = ProjectOption.objects.get_value(project=self.project, key="mail:subject_prefix")
         if not prefix:
             prefix = options.get("mail.subject-prefix")
-        return prefix
+        return str(prefix)
 
-    def should_email(self):
+    def should_email(self) -> bool:
         return True
 
-    def get_participants(self):
+    def get_participants(self) -> Mapping[User, int]:
         # TODO(dcramer): not used yet today except by Release's
         if not self.group:
-            return []
+            return {}
```

Comment:
So this used to be a list of users, and now it's a mapping of users to what? Why is it changing in this PR?

Reply:
This was probably a bug. The expected return type of `get_participants` is changing from `Mapping[User, int]` to `Mapping[ExternalProviders, Mapping[User, int]]`.